code/modules/status_system/statusEffects.dm

```diff
@@ -3125,12 +3125,14 @@
 
 		onAdd()
 			..()
+			if (locate(/datum/statusEffect/art_curse/nightmare) in src.owner.statusEffects)
+				return
```

Comment:
why not just use `hasStatus()`? I don't think that this code is hot enough to worry about microopts like this.